config/config.go

```diff
@@ -241,6 +250,7 @@ type Config struct {
 	API              ApiConfig         `yaml:"api,omitempty"`
 	Auth             AuthConfig        `yaml:"auth,omitempty"`
 	Deployment       DeploymentConfig  `yaml:"deployment,omitempty"`
+        ApiDocumentation ApiDocAnnotations `yaml:"apidoc_annotations,omitempty"`
```

Comment:
This is going to require some additions to the operator's `operator/roles/kiali_deploy/vars/main.yml` and `operator/roles/kiali_deploy/defaults/main.yml` as well as documented in the `operator/deploy/kiali_cr.yaml` file. I will add a more detailed comment in this PR once I finish this review.

Reply:
Add the documentation to this file following the same syntax/format as the rest:

https://github.com/kiali/kiali/blob/v1.1.0/operator/deploy/kiali/kiali_cr.yaml#L64

This is an example Kiali CR that has all the defaults commented out. It is formatted in way that lets a user easily create a Kiali CR (e.g. all the indentation is correct). All of these settings are alphabetically sorted - so put this new `apidoc_annotations" section after the "api" but before the "auth" section.

Make sure you set the default values in the comments (all values you see in here are the defaults - its how people will know what the defaults are; by just using this file as a reference).

Reply:
Because you are adding a new top-level group of settings, you need to add a section to this file, following the same syntax and format as the other groups - again, these are alphabetically sorted so put the new apidoc_annotations group after the "api" and before the "auth" groups:

https://github.com/kiali/kiali/blob/v1.1.0/operator/roles/kiali-deploy/vars/main.yml#L25

Reply:
Finally, you want to define the full config section (with all available settings and their default values) in here:

https://github.com/kiali/kiali/blob/v1.1.0/operator/roles/kiali-deploy/defaults/main.yml#L23

Again, alphabetically sort everything. So if I understand this PR correctly, it will look something like:

```
apidoc_annotations:
  api_spec_annotation_name: "kiali.kubernetes.io/api-spec"
  api_type_annotation_name: "kiali.kubernetes.io/api-type"

auth:
  strategy:
... the rest of the file ...
```

Reply:
As a suggestion, you should think about renaming this "apidoc_annotations" section. If additional configuration settings are needed in the future, it would be best to name this section "apidocs" - that way any new settings related to the apidocs feature can be added here even if they are not related to the annotations themselves. So something like this should be considered:

```
apidocs:
  api_spec_annotation_name: "kiali.kubernetes.io/api-spec"
  api_type_annotation_name: "kiali.kubernetes.io/api-type"
```

Or something along those lines.

Reply:
Ok, I will have a look to the operator, thanks !